src/components/error-handling/ErrorComponent.tsx

```diff
@@ -1,43 +1,45 @@
 'use client';
 
+import { Resource } from '@/schema';
 import { createSharedPathnamesNavigation } from 'next-intl/navigation';
 import { locales } from '@/locales';
+import { useEffect } from 'react';
 import { useTranslations } from 'next-intl';
 
 const { Link } = createSharedPathnamesNavigation({
   locales: [...locales.values()],
 });
 
-export default function ErrorComponent({ title, code }: { title?: string; code?: number }) {
+interface ApplicationError {
+  type: 'notFound' | 'dataEmpty' | 'dataNotLoaded' | 'resourceConfigurationInvalid' | 'unexpected';
+  resource?: Resource;
+  error?: string;
+}
+
+export default function ErrorComponent({ type, resource, error }: ApplicationError) {
   const t = useTranslations('Error');
-  let errorTitle;
-  let subTitle;
-  let codeElement;
-  if (code === 404) {
-    errorTitle = t('notFoundTitle');
-    subTitle = t('notFoundSubtitle');
-    codeElement = (
-      <h2 className="d-flex justify-content-center align-items-center gap-2 mb-4">
-        <span className="display-1 fw-bold">4</span>
-        <i className="bi bi-exclamation-circle-fill text-danger display-4"></i>
-        <span className="display-1 fw-bold bsb-flip-h">4</span>
-      </h2>
-    );
-  } else {
-    errorTitle = title ?? t('unexpectedTitle');
-    subTitle = t('genericSubtitle');
-    codeElement = code ? <span className="display-1 fw-bold">{code}</span> : <div />;
-  }
+  useEffect(() => {
+    if (type === 'dataNotLoaded' || type === 'resourceConfigurationInvalid' || type === 'unexpected') {
```

Comment:
We should probably also log the dataEmpty errors, at least by logging their message. The dataEmptyMessage is currently unused, as far as I can see.

Reply:
Good point. Forgot to add those